controllers/datanetwork_controller.go

```diff
@@ -536,8 +525,15 @@ func (r *DataNetworkReconciler) Reconcile(ctx context.Context, request ctrl.Requ
 		return ctrl.Result{}, nil
 	}
 
+	err, scope_updated := r.UpdateDeploymentScope(instance)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
 	if instance.Status.ObservedGeneration == instance.ObjectMeta.Generation && instance.Status.Reconciled {
-		return ctrl.Result{}, nil
+		if !scope_updated {
+			return ctrl.Result{}, nil
+		}
 	}
 
 	// Update scope from configuration
```

Comment:
As deployment scope was no longer updated in UpdateConfigStatus(), expect to update the comment and long here